### PR TITLE
Gui dragging polygons

### PR DESCRIPTION
**Pull request by Holandsoest**

### CODE REVIEW

File: source/common/shapes.py [hunk 1]
```diff
@@ -0,0 +1,448 @@
+#!/usr/bin/env python3
+"""Defines how to draw shapes and their bounding boxes."""
+import common.location as loc
+import tkinter
+import math
+
+object_names_array=["circle", "half circle", "square", "heart", "star", "triangle"]
+class Annotation:
```
Comment (Holandsoest) [inline source/common/shapes.py:8]: Please remove dead code

Comment (Holandsoest) [reply source/common/shapes.py:8]: No @Holandsoest i intend to keep this for future use ;-;

File: source/gui.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            self.pallet_elements.append(shapes.Square(           loc.Box(x=0, y= 8*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.Heart(            loc.Box(x=0, y= 9*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.Star(             loc.Box(x=0, y=10*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            self.pallet_elements.append(shapes.SymmetricTriangle(loc.Box(x=0, y=11*self.pallet_item_size().y, width=self.pallet_item_size().x, height=self.pallet_item_size().y), color, 'black'))
+            
+            for shape in self.pallet_elements:
+                shape.draw_shape(self, location_offset=loc.Pos())
+            self.update()
+        redraw_pallet_elements()
+        def pick_up(event):
```
Comment (Maucee) [inline source/gui.py:107]: -> return type

File: source/gui.py [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+                if event.x > shape.box.pos.x + shape.box.size.x: continue   # Right of shape.box (out of range)
+                if event.y < shape.box.pos.y: continue                      # Top of shape.box   (out of range)
+                if event.y > shape.box.pos.y + shape.box.size.y: continue   # Bottom of shape.box(out of range)
+
+                self.in_hand.append(shape)
+                shape.remove_shape(self)
+                self.list_of_canvas_shapes.remove(shape)
+                if (self.verbose_events): print (f'Shape picked up from: {shape.box}')
+                return
+            if (self.verbose_events): print ('No shape in the region')
```
Comment (Maucee) [inline source/gui.py:132]: no return at the end of the function, just in case?

File: source/gui.py [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+        def let_go(event):
```
Comment (Maucee) [inline source/gui.py:133]: -> return type

File: source/gui.py [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+                                              depth_percentage=depth_percentage)
+                    self.list_of_canvas_shapes.append(new_shape)
+                    new_shape.draw_shape(tkinter_canvas=self,
+                                         location_offset=loc.Pos(x=self.pallet_item_size().x,y=0))
+                    self.in_hand.remove(item)
+                    continue
+                
+                # Data for new shape
+                new_shape_color = self.last_color.name.lower()
+                new_shape_shape = ''        # ILLEGAL
```
Comment (Maucee) [inline source/gui.py:167]: should this be in if it is 'ILLEGAL' ?

File: source/gui.py [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+                                                  center_pos=shape.center_pos,
+                                                  size=shape.box.size, color=item)
+                    else: # Apply shape to shape 
+                        new_shape = get_new_shape(shape=item,
+                                                  center_pos=shape.center_pos,
+                                                  size=shape.box.size, color=PalletItem[shape.fill_color.upper()])
+                new_shape.draw_shape(tkinter_canvas=self,
+                                     location_offset=loc.Pos(x=self.pallet_item_size().x,y=0))
+                self.list_of_canvas_shapes.append(new_shape)
+                self.in_hand.remove(item)
```
Comment (Maucee) [inline source/gui.py:211]: Does a function in python always automatically return? without return at the end?

File: source/gui.py [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+        """Returns the size of each tool from the pallet"""
+        return loc.Size(x= 64,  #TODO: Magic number
+                        y= self.winfo_height()/(len(PalletItem)-1))
+    def canvas_size(self) -> loc.Size:
+        """Returns the usable space of the canvas (exclusive the pallet)"""
+        return loc.Size(x=self.winfo_width()-self.pallet_item_size().x, y=self.winfo_height())
+    def play_music(self, bypass_ai=False) -> None:
+        if not bypass_ai: raise RuntimeError("no u cannot do that yet :'( i dont know how the ai works ;-;")
+
+        import common.midi_creation
```
Comment (Maucee) [inline source/gui.py:283]: imports at top of document?

File: source/gui.py [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+        notes = []
+        sounds = {
+            '0' : 81,   # circle
+            '1' : 74,   # half circle
+            '2' : 119,  # square
+            '3' : 2,    # heart
+            '4' : 43,   # star
+            '5' : 30,   # triangle
+        }
+        def rgb_to_bpm(r:int, g:int, b:int) -> int: #TODO: Replace when `image_processing.py` has a better implementation
```
Comment (Maucee) [inline source/gui.py:295]: nice fix haha

File: source/gui.py [hunk 9]
```diff
@@ -0,0 +1,332 @@
+#!/usr/bin/env python3
+"""A library that covers how the GUI should respond / work
+
+Use `app = Gui()` to get started.
+
+In this version you cannot rotate & resize the shapes yet"""
+# Want to learn about Tkinter gui tool? https://youtu.be/mop6g-c5HEY It coverers about everything. ;)
+import common.shapes as shapes
+import common.location as loc
+import image_processing
```
Comment (Maucee) [inline source/gui.py:10]: double import, see further in the code

Comment (Maucee) [reply source/gui.py:10]: at 284

**Review by Maucee — APPROVED** [review]
looks good! just some comments that are easy fixes/replies

File: source/gui.py (hunk 5 above)
Comment (Holandsoest) [reply source/gui.py:167]: @Maucee, Yea... I guess it was a bit faqe, but i wanted to declare it at that depth.  
replaced it by `# Declared here as empty to be filled in later`

File: source/gui.py (hunk 3 above)
Comment (Holandsoest) [reply source/gui.py:132]: Nope it would not add value.  
See other comment

**Comment by Holandsoest** [comment]
This will merge after [issue: 19-invalid-path](https://github.com/Holandsoest/art-to-music/issues/19)

**Pull request merged**